### ISSUE TITLE
Install error with caffe-0.13  cudnn3

### ISSUE THREAD

**Issue by xiuxiazhang** [NVIDIA/caffe#45]
Hi,
     I am trying to install caffe-0.13 in order to test performance on Kepler K20 and K40 machine.
I am able to install and run BLVC/caffe  successfully.  However nvida/caffe-0.13 failed.

Error information:
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
. /usr/bin/ld: warning: libcudart.so.6.5, needed by /collect2: error: ld returned 1 exit status
usr/local/lib/libopencv_core.so, may conflict with libcudart.so.7.0
.build_release/lib/libcaffe.so: undefined reference to `cudnnSetLRNDescriptor'
.build_release/lib/libcaffe.so: undefined reference to `cudnnGetConvolutionBackwardDataWorkspaceSize'
.build_release/lib/libcaffe.so: undefined reference to `cudnnConvolutionBackwardData_v3'
.build_release/lib/libcaffe.so: undefined reference to `cudnnGetConvolutionBackwardFilterWorkspaceSize'
.build_release/lib/libcaffe.so: undefined reference to `cudnnLRNCrossChannelBackward'
.build_release/lib/libcaffe.so: undefined reference to `cudnnCreateLRNDescriptor'
.build_release/lib/libcaffe.so: undefined reference to `cudnnGetConvolutionBackwardFilterAlgorithm'
.build_release/lib/libcaffe.so: undefined reference to `cudnnLRNCrossChannelForward'
.build_release/lib/libcaffe.so: undefined reference to `cudnnDestroyLRNDescriptor'
.build_release/lib/libcaffe.so: undefined reference to `cudnnConvolutionBackwardFilter_v3'
.build_release/lib/libcaffe.so: undefined reference to `cudnnGetConvolutionBackwardDataAlgorithm'
.build_release/lib/libcaffe.so: undefined reference to `cudnnDivisiveNormalizationBackward'
.build_release/lib/libcaffe.so: undefined reference to `cudnnDivisiveNormalizationForward'
collect2: error: ld returned 1 exit status

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

It seems that libcaffe.so try to use cudnn v2 and cuda driver 6.5.  
However I've already installed cuda-7.0 and set environment for both cuda-7.0 and cudnn v3.
By the way, Does any  guy have  performance data on K20/K40 ?  
I'd like to know whether cudnn v3 has significant speed on these two platforms. 

Thanks,
Xiuxia

**Comment by borisfom** [comment]
Please use caffe-0.14 branch.

**Comment by xiuxiazhang** [comment]
Finally, I am able to compile Nvidia's branch caffe-0.14. 
But It seems that the program stuck somewhere.

I1207 11:30:08.520385  9733 net.cpp:217] label_data_1_split does not need backward computation.
I1207 11:30:08.520393  9733 net.cpp:217] data does not need backward computation.
I1207 11:30:08.520400  9733 net.cpp:260] This network produces output accuracy
I1207 11:30:08.520407  9733 net.cpp:260] This network produces output loss
I1207 11:30:08.520431  9733 net.cpp:529] Collecting Learning Rate and Weight Decay.
I1207 11:30:08.520442  9733 net.cpp:274] Network initialization done.
I1207 11:30:08.520448  9733 net.cpp:275] Memory required for data: 675621896
I1207 11:30:08.520578  9733 solver.cpp:45] Solver scaffolding done.
I1207 11:30:08.520623  9733 caffe.cpp:179] Starting Optimization
I1207 11:30:08.520665  9733 solver.cpp:269] Solving CaffeNet
I1207 11:30:08.520673  9733 solver.cpp:270] Learning Rate Policy: step
I1207 11:30:08.520690  9733 solver.cpp:314] Iteration 0, Testing net (#0)


However I could run the same configure file using BVLC/caffe  with cudnn v3. 
I noticed that, GPU memory used is much less,   may be  it stuck on somewhere before computing.
|    0      9733    C   ./build/tools/caffe                            112MiB |

[ghq@gpu201 ~]$ nvidia-smi 
Mon Dec  7 11:34:13 2015       
+------------------------------------------------------+                       
| NVIDIA-SMI 346.96     Driver Version: 346.96         |                       
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla K40m          Off  | 0000:02:00.0     Off |                  Off |
| N/A   42C    P0    63W / 235W |    170MiB / 12287MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla K40m          Off  | 0000:03:00.0     Off |                  Off |
| N/A   30C    P8    19W / 235W |     56MiB / 12287MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla K40m          Off  | 0000:83:00.0     Off |                  Off |
| N/A   27C    P8    21W / 235W |     56MiB / 12287MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla K40m          Off  | 0000:84:00.0     Off |                  Off |
| N/A   28C    P8    19W / 235W |     56MiB / 12287MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|    0      9733    C   ./build/tools/caffe                            112MiB |
+-----------------------------------------------------------------------------+

**Comment by lukeyeager** [comment]
@xiuxiazhang sorry for the late response. Did you figure out your issue? Are you still seeing a lock-up?
